app/src/main/java/net/squanchy/schedule/SchedulePageView.kt

```diff
@@ -0,0 +1,161 @@
+package net.squanchy.schedule
+
+import android.content.Context
+import android.support.design.widget.CoordinatorLayout
+import android.support.design.widget.TabLayout
+import android.support.v4.view.ViewPager
+import android.support.v7.widget.Toolbar
+import android.text.Spanned
+import android.util.AttributeSet
+import android.view.View
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
+import net.squanchy.R
+import net.squanchy.analytics.Analytics
+import net.squanchy.analytics.ContentType
+import net.squanchy.home.Loadable
+import net.squanchy.navigation.Navigator
+import net.squanchy.schedule.domain.view.Event
+import net.squanchy.schedule.domain.view.Schedule
+import net.squanchy.schedule.view.ScheduleViewPagerAdapter
+import net.squanchy.support.font.applyTypeface
+import net.squanchy.support.font.getFontFor
+import net.squanchy.support.font.hasTypefaceSpan
+import net.squanchy.support.unwrapToActivityContext
+import timber.log.Timber
+
+
+class SchedulePageView @JvmOverloads constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int = 0) : CoordinatorLayout(
+        context,
+        attrs,
+        defStyleAttr
+), Loadable {
+
+    private val viewPagerAdapter: ScheduleViewPagerAdapter
+    private val service: ScheduleService
+    private val navigate: Navigator
+    private val analytics: Analytics
+    private lateinit var progressBar: View
+    private var subscription: Disposable? = null
+
+    init {
+        val activity = unwrapToActivityContext(getContext())
+        val component = obtain(activity)
+        service = component.service()
+        navigate = component.navigator()
+        analytics = component.analytics()
+
+        viewPagerAdapter = ScheduleViewPagerAdapter(activity)
+    }
+
+    override fun onFinishInflate() {
+        super.onFinishInflate()
+
+        progressBar = findViewById(R.id.progressbar)
+
+        val viewPager = findViewById<ViewPager>(R.id.viewpager)
```

Comment:
You should prefer using synthetic — see [here](https://github.com/squanchy-dev/squanchy-android/pull/394/commits/113f75e1b4231d2467fe4f2176c47633eda133e9)

Reply:
7c5a081d70355bcd424c88a4dea65f441e651d6c